Sources/ApolloCodegenLib/ApolloCodegenConfiguration.swift

```diff
@@ -0,0 +1,140 @@
+import Foundation
+
+/// A configuration object that defines behavior for code generation.
+public struct ApolloCodegenConfiguration {
+  /// Specify the input files required for code generation.
+  public struct FileInput {
+    /// Path to the GraphQL schema file. Can be in JSON or SDL format.
+    public let schema: URL
+    /// Glob of files to search for GraphQL operations. This should be used to find queries and any client schema extensions. Defaults to `./**/*.graphql`, which will search for `.graphql` files throughout all subfolders of the folder where the script is run.
+    public let glob: String
+    /// [optional] Path to an operation id JSON map file. If specified, also stores the operation IDs (hashes) as properties on operation types. Defaults to `nil`.
+    public let operationIDs: URL?
+  }
+
+  /// Configure the folder structure of the generated files.
+  public struct FileOutput {
+    /// The location structure for the schema types files.
+    public let schemaTypes: SchemaTypesFileOutput
+    /// The location structure for the operation object files.
+    public let operations: OperationsFileOutput
+  }
+
+  /// Defines the location structure for the schema types files.
+  public struct SchemaTypesFileOutput {
+    /// Compatible dependency manager automation.
+    public enum DependencyManagerAutomation {
```

Comment:
What about
```swift
  /// Compatible dependency manager automation.
    public enum ModuleType {
      /// No module will be created for the generated schema types. 
      /// Generated files must be manually added to the main application target.
      ///
      /// Because no module is created, the generated files will be namespaced to prevent naming conflicts.
      case manuallyLinked
      /// Generates a module with a podspec file that is suitable for linking to your project using CocoaPods.
      case cocoaPods
      /// Generates a module with a cartfile that is suitable for linking to your project using Carthage.
      case carthage
      /// Generates a module with a package.swift file that is suitable for linking to your project using Swift Package Manager
      case swiftPackageManager
    }
```

Reply:
I prefer @AnthonyMDev's suggestion, personally

Reply:
👍🏻 - I agree `manuallyLinked` is more descriptive. Thanks for the extended documentation too.